metapool/prep.py

```diff
@@ -842,3 +843,84 @@ def qiita_scrub_name(name):
         the sample name, formatted for qiita
     """
     return re.sub(r'[^0-9a-zA-Z\-\.]+', '.', name)
+
+
+def _organize_by_project(pre_prep):
+    # extract project-level metadata
+
+    gb = pre_prep.groupby('project_name')
+
+    results = [gb.get_group(grp) for grp in gb.groups]
+
+    projects = []
+    data = {}
+
+    for project in results:
+        # the set of unique project_names should only be one value
+        project_name = set(project['project_name']).pop()
+
+        # assert that all values in 'contains_replicates' column for a given
+        # project are either all True or all False. Handle case-variations.
+        values = set([x.lower() == 'true' for x in
```

Comment:
Is this necessary to lower? Is there a chance to have different casing? I thought the plan was to create these via a notebook (AKA no hand creation). A more pandas way to do this would be:
`values = project_name.contains_replicates.unique()`

Reply:
It was just practicing defensive programming. I agree it's more likely than not going to be created by a notebook, but in case someone happens to one-off manually change something and it's not perfect case, it won't choke. I'm up for being more pandas!